content/influxdb/v2.2/api-guide/iot-starter/python.md

```diff
@@ -0,0 +1,743 @@
+on success. Check for any failures then return the `device_id` if the request
+was successful.
+```python
+point = Point('deviceauth') \
+        .tag("deviceId", device_id) \
+        .field('key', f'fake_auth_id_{device_id}') \
+        .field('token', f'fake_auth_token_{device_id}')
+
+ client_response = write_api.write(bucket=config.get('APP', 'INFLUX_BUCKET_AUTH'), record=point)
+
+ # write() returns None on success
+ if client_response is None:
+     return device_id
+
+ # Return None on failure
+ return None
+```
+
+## Query Bucket For Device
+After storing device information to a bucket, use the client library to query 
+for the device information.  
+
+Create a new function called `get_devices()` within `devices.py`. This function 
+will take in a `device_id` and return a list of tuples that represent the 
+records generated by the query.
+```python
+def get_device(device_id) -> {}:
+    influxdb_client = InfluxDBClient(url=config.get('APP', 'INFLUX_URL'),
+                                     token=config.get('APP', 'INFLUX_TOKEN'),
+                                     org=config.get('APP', 'INFLUX_ORG'))
+
+    # Queries must be formatted with single and double quotes correctly
+    query_api = QueryApi(influxdb_client)
+    device_id = str(device_id)
+    device_filter = f'r.deviceId == "{device_id}" and r._field != "token"'
+    flux_query = f'from(bucket: "{config.get("APP", "INFLUX_BUCKET_AUTH")}") ' \
+                 f'|> range(start: 0) ' \
+                 f'|> filter(fn: (r) => r._measurement == "deviceauth" and {device_filter}) ' \
+                 f'|> last()'
+    devices = {}
+
+    response = query_api.query(flux_query)
+
+   results = []
+       for table in response:
+           for record in table.records:
+               results.append((record.get_field(), record.get_value()))
+    return results
+```
+{{% caption %}}[@influxdata/iot-python-app ./api/devices.py get_device() line 31](https://github.com/influxdata/iot-python-api/blob/main/api/devices.py#L31){{% /caption %}}
+
+Create a `QueryApi` instance. The `QueryApi` instance queries records from a
+specified bucket using Flux. To generate the Flux query, set the bucket 
+information, range, and query filter. This query will filter on `_measurement` 
+value, `deviceauth`, and searches for any device_id that matches the passed in 
+device_id. Add clause to search for `_field`s that do not contain `token` as a 
+value.
+
+```python
+device_filter = f'r.deviceId == "{device_id}" and r._field != "token"'
+flux_query = f'from(bucket: "{config.get("APP", "INFLUX_BUCKET_AUTH")}") ' \
+           f'|> range(start: 0) ' \
+           f'|> filter(fn: (r) => r._measurement == "deviceauth" and {device_filter}) ' \
+           f'|> last()'
+```
+
+The client returns a `FluxTable`. Parse the object into a list of tuples.
+```python
+# Samples results
+[('key', 'fake_auth_id_1'), ('key', 'fake_auth_id_2')]
+```
```

Comment:
~~~suggestion
The client returns a `FluxTable`. Parse the object into a list of tuples.

```python
# Samples results
[('key', 'fake_auth_id_1'), ('key', 'fake_auth_id_2')]
```
~~~

Show the code that actually parses the response into tuples.